### PR TITLE
KAFKA-18006: Add 3.9.0 to end-to-end test (streams)

### PR DESCRIPTION
**Pull request by frankvicky**
JIRA: KAFKA-18006

Since 3.9 is released, we should add it to the e2e-test.
Follow-up of #17797 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by frankvicky** [comment]
Test result of `streams_upgrade_test.py`:
```
17:49:07 SESSION REPORT (ALL TESTS)
17:49:07 ducktape version: 0.12.0
17:49:07 session_id:       2024-11-13--001
17:49:07 run time:         52 minutes 53.820 seconds
17:49:07 tests run:        20
17:49:07 passed:           20
17:49:07 flaky:            0
17:49:07 failed:           0
17:49:07 ignored:          0
```

Test result of `streams_broker_compatibility_test.py`:
```
10:34:01 SESSION REPORT (ALL TESTS)
10:34:01 ducktape version: 0.12.0
10:34:01 session_id:       2024-11-13--001
10:34:01 run time:         9 minutes 7.208 seconds
10:34:01 tests run:        20
10:34:01 passed:           20
10:34:01 flaky:            0
10:34:01 failed:           0
10:34:01 ignored:          0
```

Test result of `streams_application_upgrade_test.py`:
```
11:03:23 SESSION REPORT (ALL TESTS)
11:03:23 ducktape version: 0.12.0
11:03:23 session_id:       2024-11-13--001
11:03:23 run time:         23 minutes 25.706 seconds
11:03:23 tests run:        17
11:03:23 passed:           17
11:03:23 flaky:            0
11:03:23 failed:           0
11:03:23 ignored:          0
```

File: build.gradle
Comment (cadonna) [inline build.gradle:1]: The following line is missing in `task testAll` property `dependsOn`:
```
':streams:upgrade-system-tests-38:test',
```

**Review by cadonna — COMMENTED** [review]
Thanks for the PR, @frankvicky !

Here my comments.

File: build.gradle
Comment (frankvicky) [reply build.gradle:1]: Thanks for pointing out.
Done 😺

**Comment by chia7712** [comment]
@frankvicky please rebase code, thanks!

**Review by chia7712 — APPROVED** [review]

File: streams/upgrade-system-tests-39/src/test/java/org/apache/kafka/streams/tests/SmokeTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    static ProcessorSupplier<Object, Object, Void, Void> printProcessorSupplier(final String topic, final String name) {
+        return () -> new ContextualProcessor<Object, Object, Void, Void>() {
+            private int numRecordsProcessed = 0;
+            private long smallestOffset = Long.MAX_VALUE;
+            private long largestOffset = Long.MIN_VALUE;
+
+            @Override
+            public void init(final ProcessorContext<Void, Void> context) {
+                super.init(context);
+                System.out.println("[3.8] initializing processor: topic=" + topic + " taskId=" + context.taskId());
```
Comment (chia7712) [inline streams/upgrade-system-tests-39/src/test/java/org/apache/kafka/streams/tests/SmokeTestUtil.java]: `3.9`

File: streams/upgrade-system-tests-39/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        kStream.to("fk-result", Produced.with(stringSerde, stringSerde));
+    }
+
+    private static <KIn, VIn, KOut, VOut> ProcessorSupplier<KIn, VIn, KOut, VOut> printProcessorSupplier(final String topic) {
+        return () -> new ContextualProcessor<KIn, VIn, KOut, VOut>() {
+            private int numRecordsProcessed = 0;
+
+            @Override
+            public void init(final ProcessorContext<KOut, VOut> context) {
+                System.out.println("[3.8] initializing processor: topic=" + topic + "taskId=" + context.taskId());
```
Comment (chia7712) [inline streams/upgrade-system-tests-39/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]: `3.9`

**Review by cadonna — APPROVED** [review]
LGTM!

Thanks, @frankvicky !

**Pull request merged**